msg/vehicle_attitude.msg

```diff
@@ -1,11 +1,17 @@
-# This is similar to the mavlink message ATTITUDE_QUATERNION, but for onboard use
+# This is similar to the mavlink message ATTITUDE_QUATERNION_COV, but for onboard use
 
-float32 rollspeed	# Bias corrected angular velocity about X body axis in rad/s
-float32 pitchspeed	# Bias corrected angular velocity about Y body axis in rad/s
-float32 yawspeed	# Bias corrected angular velocity about Z body axis in rad/s
+bool att_valid			# true if attitude is valid
+bool att_rate_valid		# true if attitude rate is valid
```

Comment:
Can we set `q` or `rollspeed` to NaN instead, to avoid adding fields?

Reply:
We could but that would mean also a change on the `vehicle_position_` msgs and on the way it interacts with the estimators right?

Reply:
>  a change on the vehicle_position_ msgs

Why?

> on the way it interacts with the estimators

In either case, if you add a valid flag or extend for NaN, the estimators need to be checked and handle the cases correctly (In case of NaN probably no change is needed)

Reply:
> Why?

So it works the same way without the `_valid` flags.

> In either case, if you add a valid flag or extend for NaN

The change would be smaller cause I would only have to add the attitude validity check. Replacing with NaN check is a bigger change.

Reply:
> So it works the same way without the _valid flags.

That would be an unrelated change. We already have both cases, some with the valid flag, others with NaN.

> The change would be smaller cause I would only have to add the attitude validity check. Replacing with NaN check is a bigger change.

The required places to add the check are exactly the same, just in one case you check for NaN and in the other you check the valid flag. Initialization can generally be kept unchanged.
You cannot add a valid flag and then just ignore it. It is not a small change in either case, it changes the fields and semantics for a very central message. Please see and handle it as such.

> Even so, can that change be left for a second PR

No, this needs to be sorted out and handled properly, otherwise the PR is incomplete and inconsistent. Ignoring a valid flag or NaN state can lead to major bugs.
Don't worry about reviews, I'll do that.